http/src/main/java/org/sapia/ubik/rmi/server/transport/http/JdkClientConnectionPool.java

```diff
@@ -15,45 +23,78 @@
  * sub-optimal implementation used only if the Jakarta HTTP client classes are
  * not in the classpath.
  * 
- * @author Yanick Duchesne
+ * @author yduchesne
  */
 public class JdkClientConnectionPool implements Connections {
-  private HttpAddress address;
-  private InternalPool pool = new InternalPool();
-
+  
+  private Category                      log    = Log.createCategory(getClass());
+  private HttpAddress                   address;
+  private Set<JdkRmiClientConnection>   active = Collections.newSetFromMap(new ConcurrentHashMap<>());
+  private InternalPool                  pool;
+  private JdkRmiClientConnectionFactory factory;
+  
+  /**
+   * @param address the address of the target server.
+   * @param factory the {@link JdkRmiClientConnectionFactory} to use.
+   * @param maxSize the pool's maximum size.
+   */
+  public JdkClientConnectionPool(HttpAddress address, JdkRmiClientConnectionFactory factory, int maxSize) {
+    this.address = address;
+    this.factory = factory;
+    this.pool = new InternalPool(maxSize);
+  }
+  
   /**
    * @param address
    *          the address of the target server.
+   * @param maxSize
+   *          the pool's maximum size.
    */
-  public JdkClientConnectionPool(HttpAddress address) {
-    this.address = address;
+  public JdkClientConnectionPool(HttpAddress address, int maxSize) {
+    this(address, () -> { 
+      JdkRmiClientConnection conn =  new JdkRmiClientConnection();
```

Comment:
Special to have a creation created in constructor... is it for test purposes?

Reply:
No the syntax corresponds to an anonymous implementation of a functional interface (JdkRmiClientConnectionFactory). It's as if we add a default impl. as an anonymous class.

Reply:
And yes that factory has been introduced as a hook in the context of testing. Outside of that context, the anonymous default impl. is used.